Userland/Applications/ImageViewer/main.cpp

```diff
@@ -346,13 +360,19 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     })));
     TRY(help_menu->try_add_action(GUI::CommonActions::make_about_action("Image Viewer", app_icon, window)));
 
+    window->show();
+
+    // We must do this here and not any earlier, as we need a visible window to call FileSystemAccessClient::Client::request_file_read_only_approved();
     if (path != nullptr) {
-        widget->load_from_file(path);
+        auto result = FileSystemAccessClient::Client::the().request_file_read_only_approved(window, path);
```

Comment:
Can't we just `TRY` here?

Reply:
`result` will be an error ("Unknown error") if the user denies access, and I feel like the actual error here isn't actually required.

This is a pattern that most other apps seem to follow, so I think we should be consistent here.

https://github.com/SerenityOS/serenity/blob/cbe090170629b7e207153ad98c0cdec4762e77b0/Userland/Applications/TextEditor/main.cpp#L79-L85

https://github.com/SerenityOS/serenity/blob/cbe090170629b7e207153ad98c0cdec4762e77b0/Userland/Applications/PDFViewer/main.cpp#L49-L51

https://github.com/SerenityOS/serenity/blob/cbe090170629b7e207153ad98c0cdec4762e77b0/Userland/Applications/HexEditor/main.cpp#L56-L58

Reply:
There is room for improvements in a lot of place in the system, so I'm not sure if consistency is really a concern. HexEditor and TextEditor are both a bit different, as they handle nonexistent files. But if you run `PDFViewer well_hello_friends` it just returns without a message, which is quite sad.
By the way, no user action is required with `request_file_read_only_approved` so no decline possible. It allows displaying all error that are not handled by FSAS (checking that the file is not a directory nor a device).

Reply:
I see, makes sense. I'll make this change when I'm home later on